src/wmtk/TetMeshOperationExecutor.cpp

```diff
@@ -454,6 +382,15 @@ Tuple TetMesh::TetMeshOperationExecutor::split_edge()
         long t_f3; // next t1
         long t_f4; // next t2
 
+        // for return tuple
+        // return_flag == true means this is the tet for the tuple to return
+
+        bool return_flag = false;
```

Comment:
I would really prefer if this whole `return_tuple` code is moved into its own for-loop. You are doing two complicated things here within one loop and this is confusing to read. If it is not necessary, try to split them up. You could even put all the `return_tuple` code in its own function named `construct_return_tuple()` or something similar.